### PR TITLE
Make configurable the initial and max size of `InternalThreadLocalMap…

### PR DESCRIPTION
**Pull request by fenik17**
….stringBuilder()`

Motivation:
In some cases of using an `InternalThreadLocalMap.stringBuilder()`, the `StringBuilder`s size can often exceed the exist limit (1024 bytes). This can lead to permanent memory reallocation.

Modifications:
Add custom properties for the initial capacity and maximum size (after which the `StringBuilder`s capacity will be reduced to the initial capacity).

Result:
An `InternalThreadLocalMap.stringBuilder()`s initial and max size is configurable. Fixes [#7092].

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Scottmitch PTAL as well

File: common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java [hunk 1]
```diff
@@ -163,17 +177,16 @@ public int size() {
...
-            stringBuilder = new StringBuilder(stringBuilderCapacity);
-        } else {
-            if (stringBuilder.capacity() > stringBuilderCapacity) {
-                stringBuilder.setLength(stringBuilderCapacity);
-                stringBuilder.trimToSize();
-            }
-            stringBuilder.setLength(0);
+        StringBuilder sb = stringBuilder;
+        if (sb == null) {
+            return stringBuilder = new StringBuilder(STRING_BUILDER_INITIAL_SIZE);
```
Comment (netkins) [inline common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java]: MAJOR Extract the assignment out of this expression. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)

File: common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java [hunk 2]
```diff
@@ -35,10 +37,22 @@
...
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(InternalThreadLocalMap.class);
+
     private static final int DEFAULT_ARRAY_LIST_INITIAL_CAPACITY = 8;
+    private static final int STRING_BUILDER_INITIAL_SIZE;
+    private static final int STRING_BUILDER_MAX_SIZE;
 
     public static final Object UNSET = new Object();
 
+    static {
+        STRING_BUILDER_INITIAL_SIZE = SystemPropertyUtil.getInt("io.netty.threadLocalStringBuilderInitialSize", 1024);
```
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java]: what about breaking out the name space more clearly: `io.netty.threadLocalMap.stringBuilder.initialSize` ... @normanmaurer - wdyt?

Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java]: @Scottmitch +1... @fenik17 please address and I will pull in once done

Comment (fenik17) [inline common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java]: @normanmaurer addressed.

**Comment by Scottmitch** [comment]
4.1 (e249b453a0) 4.0 (61d3d944e2)

**Pull request closed without merging**